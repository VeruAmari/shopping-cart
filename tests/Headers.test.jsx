/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import Heading from '../src/components/sub/Headers';

describe('Heading', () => {
  it('renders', () => {
    render(<Heading level="1">Level 1 Heading</Heading>);
    expect(screen.getByText('Level 1 Heading').textContent).toMatch(/level/i);
  });
  it('correctly handles heading levels', () => {
    render(
      <>
        <div data-testid="parent">
          <Heading level="1">Heading</Heading>
        </div>
        <div data-testid="parent-2">
          <Heading level="2">Heading</Heading>
        </div>
        <div data-testid="parent-3">
          <Heading level="6">Heading</Heading>
        </div>
      </>,
    );
    expect(screen.getByTestId('parent').firstChild).toContainHTML('</h1>');
    expect(screen.getByTestId('parent-2').firstChild).toContainHTML('</h2>');
    expect(screen.getByTestId('parent-3').firstChild).toContainHTML('</h6>');
  });

  it('correctly sets margins', () => {
    render(
      <>
        <Heading margin="2">Heading with a margin of 2</Heading>
        <Heading margin="0">Heading with a margin of 0</Heading>
        <Heading margin=".2">Heading with a margin of 0.2</Heading>
        <Heading>Heading with a default margin</Heading>
        <Heading margin="a">Heading margin ignores a string value</Heading>
        <Heading margin={null}>Heading margin ignores a null value</Heading>
      </>,
    );
    expect(screen.getByText('Heading with a margin of 2').style.margin).toMatch(
      '2rem',
    );
    expect(screen.getByText('Heading with a margin of 0').style.margin).toMatch(
      '0',
    );
    expect(
      screen.getByText('Heading with a margin of 0.2').style.margin,
    ).toMatch('.2rem');

    expect(
      screen.getByText('Heading with a default margin').style.margin,
    ).toMatch('.5rem');
    expect(
      screen.getByText('Heading margin ignores a string value').style.margin,
    ).toMatch('.5rem');
    expect(
      screen.getByText('Heading margin ignores a null value').style.margin,
    ).toMatch('.5rem');
  });
});
