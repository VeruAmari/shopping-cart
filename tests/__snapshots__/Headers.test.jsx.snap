// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heading > matches snapshot 1`] = `
<h1
  class="Headers__Wrapper-sc-o55uf2-0 ceolFo"
  style="margin: 2rem;"
>
  Snapshot
</h1>
`;
